### DISCUSSION TITLE
Support microcks-testcontainers-dotnet?

### DISCUSSION THREAD

**Discussion by thangchung** [microcks/microcks#1102] [Q&A] [answered]
Context: we come from the .NET community and would like to have the ability to use microrocks with testcontainers.

We just came around in some communities like Java, Go, and Node and saw there are native supports to run microrocks with testcontainers, and see that there are a lot of benefits on integration tests, and mocking some external services (via OpenAPI or even AsyncApi specs). My question is do we have a plan to support it for .NET/C#? Thank you very much.

**Comment by lbroudoux** [comment, marked as answer]
Hi @thangchung 

At the moment there's no .Net/C# skills among the maintainers but it would be awesome to have a Microcks testcontainers module for this stack!

If this is something you'd be able and would like to contribute, we'd be more than happy to welcome your contribution under the Microcks umbrella. Do not hesitate to join us on Discord (https://microcks.io/discord-invite) where we have a #testcontainers channel to discuss features, ideas an so on. We could discuss together the existing design and the application to the .Net stack. 

Cheers,

**Reply by lbroudoux** [reply]
Hey @thangchung

Are there any folks on your team who'd like to contribute to a cool project? We'd be more than happy to guide someone through becoming a contributor to this .Net module!

**Reply by SebastienDegodez** [reply]
WIP: [feat: Add Microcks Module · SebastienDegodez/testcontainers-dotnet](https://github.com/SebastienDegodez/testcontainers-dotnet/commit/3a951c12fb13cc091f9e414208a903750d5b63c1)

**Reply by lbroudoux** [reply]
Awesome!! 🎉  Do you already have something running? Can we help with something?

**Reply by SebastienDegodez** [reply]
Hello, if I compare with your java code, testContractTestingFunctionalityWithOAuth2 & testContractTestingFunctionality are not yet implemented.

**Reply by lbroudoux** [reply]
Excellent! `testContractTestingFunctionalityWithOAuth2` was just written this morning BTW 😉 That's very promising!  Let me know if I can help on something.

**Reply by lbroudoux** [reply]
Are you writing this library for personal or professional use? I'd be super interested to learn more on how you're using Microcks!

**Reply by SebastienDegodez** [reply]
I think it will be a help in my (professional) project in AXA France, we have Java and C#, with old services in SOAP, others in REST and also Kafka. 

It will allow me to standardize our methods.

My first idea would be to host it on a dev environment (k8s), I would like to explore this path one day.

Other tech leads are already very interested since I introduced them to Microcks.
